### ISSUE TITLE
Bug in image extractor handling

### ISSUE THREAD

**Issue by rlopezcoto** [cta-observatory/cta-lstchain#293]
Hi all,
I open this issue for discussion, I will open a separate PR for the fixing. @FrancaCassol, @moralejo and I have been investigating the light collection efficiency derived using muon analysis. It seems that we have found a bug on how the selected image extractor is passed to the analysis chain. At the moment, the image extractor is defined in:
https://github.com/cta-observatory/cta-lstchain/blob/921c5cc65d4577a08fa4db7bbe197705ae42bdbb/lstchain/data/lstchain_standard_config.json#L100

the script of data analysis calls:
https://github.com/cta-observatory/cta-lstchain/blob/921c5cc65d4577a08fa4db7bbe197705ae42bdbb/lstchain/scripts/lstchain_data_r0_to_dl1.py#L67

that calls the `LSTCameraCalibrator`
https://github.com/cta-observatory/cta-lstchain/blob/921c5cc65d4577a08fa4db7bbe197705ae42bdbb/lstchain/reco/dl0_to_dl1.py#L192

here we pass the variable `image_extractor`
https://github.com/cta-observatory/cta-lstchain/blob/921c5cc65d4577a08fa4db7bbe197705ae42bdbb/lstchain/reco/dl0_to_dl1.py#L194

but the one `LSTCameraCalibrator` expects is `extractor_product`:
https://github.com/cta-observatory/cta-lstchain/blob/921c5cc65d4577a08fa4db7bbe197705ae42bdbb/lstchain/calib/camera/calibrator.py#L20

The fix is just to change `image_extractor` -> `extractor_product` in https://github.com/cta-observatory/cta-lstchain/blob/921c5cc65d4577a08fa4db7bbe197705ae42bdbb/lstchain/reco/dl0_to_dl1.py#L194

I'm currently running tests,  I'll make the PR afterwards. Any opinions?

### Consequences:
- All the *real data* that has been processed up to now are suffering from this bug, so any signal extractor that you have been using was not passed to the analysis. I checked this using the data processed by @morcuended using different signal extractors and results are identical.
- The data was processed using the default image extractor defined in `extractor_product`: `NeighborPeakWindowSum`, with its default initialization ( window_shift: 3, window_width: 7). This is specially bad because of time misalignment in neighbor modules (@FrancaCassol can comment further on that).

**Comment by kosack** [comment]
Do you use your own config file system, or are you using the ctapipe one? If the ctapipe one (i.e your scipt inherits from `ctapipe.core.Tool`), you are construction your CameraCalibrator incorrectly: you can just do something like this:

```py3
self.image_extractor = self.add_component(
            ImageExtractor.from_name(
                self.image_extractor_type,  # where this is some config variable with a string name
                parent=self,
                subarray=self.event_source.subarray,
            )
  )
 self.calibrate = self.add_component(
            CameraCalibrator(parent=self, image_extractor=self.image_extractor)
 )
```

It will be configured automatically.  note that the add_component wrapper function is only used in the development version of ctapipe, but isn't there in older version (it allows the configuration to be tracked and validated against what the user requested, so would avoid this bug you have).

**Comment by FrancaCassol** [comment]
Hi @kosack,
yes,  the script is not based on the ctapipe.core.Tool and it uses its own config file system. We should indeed move as soon as possible to a ctapipe.core.Tool and with the occasion to update the code to the master of ctapipe

**Comment by rlopezcoto** [comment]
Thanks for the comment @kosack. @FrancaCassol is right, but we would probably need a new stable release of ctapipe for it. I've been also discussing with @MaxNoe, are you guys planning to do a release soon?

**Comment by rlopezcoto** [comment]
Solved in #297
